app/src/main/java/com/infomaniak/mail/ui/main/search/SearchViewModel.kt

```diff
@@ -0,0 +1,181 @@
+/*
+ * Infomaniak kMail - Android
+ * Copyright (C) 2023 Infomaniak Network SA
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package com.infomaniak.mail.ui.main.search
+
+import android.util.Log
+import androidx.lifecycle.*
+import com.infomaniak.lib.core.models.ApiResponse
+import com.infomaniak.mail.data.api.ApiRepository
+import com.infomaniak.mail.data.cache.mailboxContent.FolderController
+import com.infomaniak.mail.data.cache.mailboxContent.MessageController
+import com.infomaniak.mail.data.cache.mailboxContent.ThreadController
+import com.infomaniak.mail.data.cache.mailboxInfo.MailboxController
+import com.infomaniak.mail.data.models.Folder
+import com.infomaniak.mail.data.models.Thread
+import com.infomaniak.mail.data.models.Thread.ThreadFilter
+import com.infomaniak.mail.ui.main.search.SearchFragment.VisibilityMode
+import com.infomaniak.mail.utils.AccountUtils
+import com.infomaniak.mail.utils.SearchUtils
+import com.infomaniak.mail.utils.SearchUtils.convertToSearchThreads
+import io.sentry.Sentry
+import kotlinx.coroutines.*
+
+class SearchViewModel : ViewModel() {
+
+    private val searchQuery = MutableLiveData<String>()
+    private val _selectedFilters = MutableLiveData<MutableSet<ThreadFilter>>()
+    private inline val selectedFilters get() = _selectedFilters.value ?: mutableSetOf()
+    val visibilityMode = MutableLiveData(VisibilityMode.RECENT_SEARCHES)
+
+    /** It is simply used as a default value for the api */
+    private lateinit var dummyFolderId: String
+    private var selectedFolder: Folder? = null
+    private var resourceNext: String? = null
+    private var resourcePrevious: String? = null
+
+    private var fetchThreadsJob: Job? = null
+
+    val searchResults = observeSearchAndFilters().switchMap { (query, filters) ->
+        val searchQuery = if (isLengthTooShort(query)) null else query
+        fetchThreads(searchQuery, filters)
+    }
+    val folders = liveData(viewModelScope.coroutineContext + Dispatchers.IO) { emit(FolderController.getFolders()) }
+
+    val hasNextPage get() = !resourceNext.isNullOrBlank()
```

Comment:
It is only used for the negation (e.g. `if (!hasNextPage)`), so we could get rid of this double negation